internal/infra/run.go

```diff
@@ -326,6 +335,11 @@ func runContainers(ctx context.Context, params RunParams, api *server.API) error
 			return err
 		}
 
+		err = pullImage(ctx, cli, params.CollectorImage)
```

Comment:
I would skip the pulling unless EnableOpenTelemetry is set, that would cause it to download for our smoke tests which don't need it.